### ISSUE TITLE
Feature request: disable heater when issue with temp probe is detected/suspected

### ISSUE THREAD

**Issue by riley56m** [reef-pi/reef-pi#1576]
Hello, after testing a few different DS18B20 temperature probes, I noticed one sensor that had failed on me would always give reef-pi a reading on 0 when it is plugged in, which in turn causes the heater to run full blast indefinitely. This would certainly kill your tank if you don't catch it quickly, so I pose in the default action for reef-pi when it reads abnormal temperatures like that is to disable the heater.

I propose adding some kind of catch to that functionality, such that if temperature suddenly changes drastically, or maybe even if temperatures are outside of livable range, the heater is deactivated entirely. if the reef tank is below 5 degrees everything is probably already dead at that point anyway, no reason to heat it up. if the temp probe is constantly detecting 0, I would prefer my heater is disabled until I can investigate, rather than it runs full blast. This could save many a tank that has a temp probe fail.

Thanks!

**Comment by schwabix-1311** [comment]
A smart solution could continue heating with the pattern of past 24h, the data does exist. This won't be too far off in most situations.

**Comment by ranjib** [comment]
we have thought about this in past and the observation was most reef tanks uses a present heater to keep temperature around 78. External controller provide extra protection or can minimize the temperature swing (heater thermister  have higher hysteresis/allowed temperature swing). Hence in case of temperature sensors it was preferred to keep the equipment in its last informed state. 
But i can see cases where folks would like to choose a different action.

**Comment by riley56m** [comment]
>the observation was most reef tanks uses a present heater to keep temperature around 78

Yes I agree, that is one potential failsafe. My concern with that, is that that internal temp sensor on the heater itself is one of the first things to fail hardware wise, resulting in a cooked tank. I think it's generally recommended to replace your heater every X months, which is $$/ unrealistic. If people use old heaters or used heaters there is a higher probability of failure.

This is why many reefers choose to connect an external controller, as those are much less likely to fail, and have the added benefit of minimizing temp swing by being more accurate and sensitive.

My position is it is safer, and therefore we should always assume the internal temp regulator on the heater itself has failed, the heater is persistently on, and the external controller is required to turn it off. If the external temp sensor fails, and the heater is not disabled your tank could be cooked in an hour or less depending on size. Safest option is always to disable heat, critters can withstand cold a lot better than hot.
